File: InternetBulletin.Shared/DTOs/AddPostDTO.cs
// *********************************************************************************
//	<copyright file="AddPostDTO.cs" company="Personal">
//		Copyright (c) 2025 Personal
//	</copyright>
// <summary>Add post dto.</summary>
// *********************************************************************************

namespace InternetBulletin.Shared.DTOs
{
    /// <summary>
    /// Add post dto.
    /// </summary>
    public class AddPostDTO
    {
        /// <summary>
		/// Gets or sets the post title.
		/// </summary>
		/// <value>
		/// The post title.
		/// </value>
        public string PostTitle { get; set; }

        /// <summary>
		/// Gets or sets the content of the post.
		/// </summary>
		/// <value>
		/// The content of the post.
		/// </value>
		public string PostContent { get; set; }

        /// <summary>
        /// Gets or sets the post created by.
        /// </summary>
        /// <value>
        /// The post created by.
        /// </value>
        public string PostCreatedBy { get; set; }

    }
}

File: InternetBulletin.Web/Models/AddPostDto.js
/**
 * AddPostDtoModel class to represent the data for adding a post.
 */
class AddPostDtoModel {
	/**
	 * Gets or sets the post title.
	 * @type {string}
	 */
	postTitle = "";

	/**
	 * Gets or sets the content of the post.
	 * @type {string}
	 */
	postContent = "";

	/**
	 * Gets or sets the post created by.
	 * @type {string}
	 */
	postCreatedBy = "";

	/**
	 * Creates an instance of AddPostDtoModel.
	 * @param {string} PostTitle The title of the post.
	 * @param {string} PostContent The content of the post.
	 * @param {string} PostCreatedBy The user who created the post.
	 */
	constructor ( PostTitle, PostContent, PostCreatedBy ) {
		this.postTitle = PostTitle;
		this.postContent = PostContent;
		this.postCreatedBy = PostCreatedBy;
	}
}

export default AddPostDtoModel;
